app/controllers/access_tokens_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class AccessTokensController < ApplicationController
+  def post
+    render problem: { error: 'invalid_request' }, status: :bad_request and return unless contains_required_params?
+    render problem: { error: 'unsupported_grant_type' }, status: :bad_request and return unless supported_grant_type?
+    render problem: { error: 'invalid_grant' }, status: :bad_request and return if current_user.nil?
+
+    access_token = current_user.access_tokens.build access_token_params
+
+    if access_token.save
+      set_disble_caching_headers
+
+      render json: AccessTokenResource.new(access_token)
+    elsif access_token.errors.key? :scope
+      render problem: { error: 'invalid_scope' }, status: :bad_request
+    else
+      render problem: { errors: [{ name: 'access_token', reason: 'can not be created successfully' }] },
+             status:  :internal_server_error
+    end
+  end
+
+  private
+
+  def current_user
+    @current_user ||= User.find_by(email: params[:username])&.authenticate params[:password]
+  end
+
+  def access_token_params
+    params.permit :scope
+  end
+
+  def contains_required_params?
+    params.keys.to_set.superset? %w[grant_type username password].to_set
+  end
```

Comment:
提案: 僕ならどう書くだろう? と考えてみた。文字数はちょっと多いけど、やりたいことをストレートに表現できている気がするがどうだろうか。

```suggestion
  def contains_required_params?
    params.require(%w[grant_type username password])
  rescue ActionController::ParameterMissing
    false
  end
```

Reply:
なるほど、ありですね。

Reply:
集合を利用した方法は確かにわかりづらかったので、#all?を利用する方法に変更しました。今回はこれで行こうと思います。

#requireを利用する場合、このメソッドはキーの存在のみだけでなく、値が#present?かどうかも見てくれるようなので、それが必要であれば、そちらを利用するかもしれませんね。でも、他に方法がないならともかく、ここで例外を使うのはなんだかなー、と少しだけモヤモヤを感じています。
https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require

Reply:
`#all?` を使った今のコードも十分にシンプルでよさそうです。